core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1119,10 +1119,13 @@ object TestUtils extends Logging {
   }
 
   def saslConfigs(saslProperties: Option[Properties]): Properties = {
-    saslProperties match {
+    val result = saslProperties match {
       case Some(properties) => properties
       case None => new Properties
     }
+    if (JaasTestUtils.isIBMJdk)
+      result.put(KafkaConfig.SaslKerberosServiceNameProp, JaasTestUtils.serviceName)
```

Comment:
An alternative would be to define a method in `JaasTestUtils` to get additional properties. That would keep JRE checks and knowledge of service name restrictions within `JaasTestUtils`.